pyvista/core/filters/image_data.py

```diff
@@ -1604,3 +1606,287 @@ def _update_and_get_output():
+        >>> segmented_grid.cell_data["Data"] = [
+        ...     0,
+        ...     0,
+        ...     0,
+        ...     1,
+        ...     0,
+        ...     1,
+        ...     1,
+        ...     2,
+        ...     0,
+        ...     0,
+        ...     0,
+        ...     0,
+        ... ]
+        >>> segmented_grid.plot(show_edges=True)
+
+        Label the connected regions. The cells with a `0` value are excluded from the
+        connected regions and labelled with `0`. The remaining cells define 3 different
+        regions that are labelled by decreasing size.
+
+        >>> connected = segmented_grid.label_connectivity()
+        >>> pl = pv.Plotter()
+        >>> _ = pl.add_mesh(connected.threshold(0.5), show_edges=True)
+        >>> _ = pl.add_mesh(
+        ...     connected.threshold(0.5, invert=True),
+        ...     show_edges=True,
+        ...     opacity=0.5,
+        ... )
+        >>> pl.show()
+
+        Exclude the cell with a `2` value.
+
+        >>> connected = segmented_grid.label_connectivity(
+        ...     scalar_range=[1, 1]
+        ... )
+        >>> pl = pv.Plotter()
+        >>> _ = pl.add_mesh(connected.threshold(0.5), show_edges=True)
+        >>> _ = pl.add_mesh(
+        ...     connected.threshold(0.5, invert=True),
+        ...     show_edges=True,
+        ...     opacity=0.5,
+        ... )
+        >>> pl.show()
+
+        Label all connected regions with a constant value.
+
+        >>> connected = segmented_grid.label_connectivity(
+        ...     label_mode='constant', constant_value=10
+        ... )
+        >>> pl = pv.Plotter()
+        >>> _ = pl.add_mesh(connected.threshold(0.5), show_edges=True)
+        >>> _ = pl.add_mesh(
+        ...     connected.threshold(0.5, invert=True),
+        ...     show_edges=True,
+        ...     opacity=0.5,
+        ... )
+        >>> pl.show()
+
+        Label only the regions that include seed points, by seed order.
+
+        >>> points = [(2, 1, 0), (0, 0, 1)]
+        >>> connected = segmented_grid.label_connectivity(
+        ...     extraction_mode='point_seeds', point_seeds=points
+        ... )
+        >>> pl = pv.Plotter()
+        >>> _ = pl.add_mesh(connected.threshold(0.5), show_edges=True)
+        >>> _ = pl.add_mesh(
+        ...     connected.threshold(0.5, invert=True),
+        ...     show_edges=True,
+        ...     opacity=0.5,
+        ... )
+        >>> pl.show()
+
+        """
+        # Get a copy of input to not overwrite data
+        input_mesh = self.copy()  # type: ignore[attr-defined]
+
+        if scalars is None:
+            set_default_active_scalars(input_mesh)
+        else:
+            input_mesh.set_active_scalars(scalars)
+        # Make sure we have point data (required by the filter)
+        field, scalars = input_mesh.active_scalars_info
+
+        if field == FieldAssociation.CELL:
+            # Convert to point data
+            input_mesh = input_mesh.cells_to_points(scalars=scalars)
```

Comment:
Minor optimization, a copy was already made earlier
```suggestion
            input_mesh = input_mesh.cells_to_points(scalars=scalars, copy=False)
```